OrbitCore/LinuxTracingSession.cpp

```diff
@@ -23,6 +26,27 @@ void LinuxTracingSession::RecordHashedCallstack(
   hashed_callstack_buffer_.push_back(std::move(hashed_call_stack));
 }
 
+void LinuxTracingSession::SetStringManager(
+    std::shared_ptr<StringManager>& string_manager) {
+  string_manager_ = string_manager;
+}
+
+void LinuxTracingSession::SendKeyAndString(
+    uint64_t key, const std::string& str) {
+  KeyAndString key_and_string;
+  key_and_string.key = key;
+  key_and_string.str = str;
+  // TODO: This is not atomic, we might end up sending multiple keys
+  // maybe change it to AddIfDoesNotExists()
+  if (!string_manager_->Exists(key)) {
+    std::string message_data = SerializeObjectBinary(key_and_string);
+    // TODO: Remove networking from here.
+    GTcpServer->Send(Msg_KeyAndString, message_data.c_str(),
```

Comment:
maybe pass GTcpServer it to constructor - it will slightly reduce numbers of globals used..

Reply:
Added a TODO.

Reply:
Actually, will change this.

Reply:
Done.